tests/integration/test_dynamodb.py

```diff
@@ -397,23 +412,24 @@ def test_dynamodb_stream_stream_view_type(self):
 
         # get records
         record = ddbstreams.get_records(ShardIterator=response['ShardIterator'])
-        # assert stream_view_type of the table
-        dynamodb.update_item(TableName=table_name,
-            Key={'Username': {'S': 'Fred'}},
-            UpdateExpression='set S=:r',
-            ExpressionAttributeValues={
-                ':r': {'S': 'Fred_Modified'}
-            },
-            ReturnValues='UPDATED_NEW'
-        )
-        self.assertEquals('KEYS_ONLY', result['StreamDescription']['StreamViewType'])
-        # assert stream_view_type of records inserted into the table
+
+        # assert stream_view_type of records forwarded to the stream
         self.assertEquals('KEYS_ONLY', record['Records'][0]['dynamodb']['StreamViewType'])
-        updated_record = ddbstreams.get_records(ShardIterator=response['ShardIterator'])
-        # assert oldImage not in the updated record info written into the stream
-        self.assertNotIn('OldImage', updated_record['Records'][1]['dynamodb'])
-        # assert newImage not in the updated record info written into the stream
-        self.assertNotIn('NewImage', updated_record['Records'][1]['dynamodb'])
+        self.assertEquals('KEYS_ONLY', record['Records'][1]['dynamodb']['StreamViewType'])
+        self.assertEquals('KEYS_ONLY', record['Records'][2]['dynamodb']['StreamViewType'])
+        # assert Keys present in the record for all insert, modify and delete events
+        self.assertEquals({'Username': {'S': 'Fred'}}, record['Records'][0]['dynamodb']['Keys'])
```

Comment:
nitpick: We could run the assertions in lines 421-431 a `for i in range(3) ...` loop.